File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
         vector<int> twoSum;
        int first = 0;
        int last = numbers.size() - 1;
        while(first < last) {
            int sum = numbers[first] + numbers[last];
            if(sum == target) {
                twoSum.push_back(first+1);
                twoSum.push_back(last+1);
                return twoSum;
            }
            if(sum < target) {
                first++;
            }
            if(sum > target) {
                last--;
            }
        }
        return twoSum;
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        res = [-1, -1]
        if n < 2:
            return res
        map = {}
        res = [-1, -1]
        for i in range(n):
            diff =  target - nums[i]
            if diff in map and map[diff] != i:
                res[0] = map[diff]
                res[1] = i
            map[nums[i]] = i
        return res